drivers/flash/soc_flash_silabs_series2.c

```diff
@@ -24,10 +24,16 @@ LOG_MODULE_REGISTER(flash_silabs);
 
 struct flash_silabs_data {
 	struct k_sem mutex;
```

Comment:
I believe it should be a `k_mutex`.

Reply:
> I believe it should be a `k_mutex`.

Why is that? (I agree about the variable name though - it should be consistent with the type, or even better something that describes its purpose rather than makes a reference to the type)

Ever since watching Andy's [presentation on the topic of synchronization primitives](https://www.youtube.com/watch?v=p8OgqVQxklo), I've had the impression that `k_sem` is often a better choice than `k_mutex`.

Reply:
Another point in favor of `k_sem` from the #kernel channel today:
https://discord.com/channels/720317445772017664/884851987508781066/1344348090982731908

Reply:
Would perhaps simply `lock` be a better name?

Reply:
In fact, I have no strong opinion:
  - Involving Priority Inheritance here is probably a waste of resources (but this argument contradict "premature optimisation is evil")
  - From a semantic point of view, this should be a mutex. If someone wants to prove some realtime requirements, he will be happy to have the distinction between semaphore and mutexes (but this argument contradict "YAGNI").


(I believe the Andy's point was only about k_mutex in ISR, which is not the case here)